trainer_core.py

```diff
@@ -0,0 +1,77 @@
+import pandas as pd
+import numpy as np
+import time
+import keras
+from keras.layers import Input, Embedding, LSTM, Dense, Conv2D
+from keras.models import Model, Sequential
+from keras.layers.core import Dense,Activation
+from keras.optimizers import Adam
+from keras.callbacks import EarlyStopping
+from keras.preprocessing.sequence import pad_sequences
+from keras.layers import LSTM, Dropout, Input
+import keras.preprocessing.text
+import tensorflow as tf
+import operator
+import sys
+
+INPUT_TEXT_LENGTH = 50
+
+def load_colors():
+    with open('colors.txt') as colors_file:
+        return list(map(lambda color: color.replace('\n', ''), colors_file.readlines()))
+
+def load_letters():
+    letter_nums = list(range(ord('A'), ord('H') + 1))
+    return list(map(lambda i: chr(i), letter_nums))
```

Comment:
Prime candidate for an eta reduction. (I just wanted to type that, you don't actually have to change it.)